lib/Infer/AliveDriver.cpp

```diff
@@ -342,18 +342,26 @@ synthesizeConstantUsingSolver(tools::Transform &t,
   return SynthesisResult;
 }
 
-souper::AliveDriver::AliveDriver(Inst *LHS_, Inst *PreCondition_, InstContext &IC_)
+souper::AliveDriver::AliveDriver(Inst *LHS_, Inst *PreCondition_, InstContext &IC_, std::vector<Inst *> ExtraInputs)
     : LHS(LHS_), PreCondition(PreCondition_), IC(IC_) {
   IsLHS = true;
   InstNumbers = 101;
   //FIXME: Magic number. 101 is chosen arbitrarily.
   //This should go away once non-input variable names are not discarded
 
+  // If alive ever supports src and tgt with different number of arguments in future, this can go away.
+  for (auto Extra : ExtraInputs) {
+    if(!translateAndCache(Extra, LHSF, LExprCache)) {
```

Comment:
put a space after the "if"

what is the consequence of this message? what is the souper user intended to learn from it?

Reply:
It is a hack to make sure that src and tgt have the same number of arguments.
We want to show that src(x) = x, and tgt(x, y) = x are same in this case.
This hack adds the extra argument to src.  
The comment is to help diagnose potential issues because of adding magic arguments.

Reply:
but I am asking what is the consequence of the message appearing. 
it sounds like you are saying this would indicate a fatal flaw in souper.
if this is correct, you need to change this to either an assertion or else an llvm::report_fatal_error()